rr_iarrc/src/finish_line_watcher/finish_line_watcher.cpp

```diff
@@ -110,21 +193,20 @@ int main(int argc, char** argv) {
     nhp.param("blockWheels_height", blockWheels_height, 800);
     nhp.param("blockBumper_height", blockBumper_height, 800);
 
+    nhp.param("publish_when_detected", publish_when_detected, true);
+
+    nhp.param("min_contour_area", min_contour_area, 5.0);
+    nhp.param("angle_cutoff", angle_cutoff, 20.0);
+    nhp.param("width_cutoff", width_cutoff, 275.0);
+
     ROS_INFO("Finish line watching %s", img_topic.c_str());
 
-    Subscriber img_saver_sub = nh.subscribe(img_topic, 1, ImageCB);
+    ros::Subscriber img_saver_sub = nh.subscribe(img_topic, 1, ImageCB);
 
     crosses_pub = nh.advertise<std_msgs::Int8>("finish_line_crosses", 1);
+    debug_pub = nh.advertise<sensor_msgs::Image>("debug", 1);
```

Comment:
Use `nhp` instead of `nh` to use private namespace (aka the node name)

Reply:
Should I change the subscriber to use `nph` too and remove the `nh` variable ?

Reply:
The reason to use the private NodeHandle in this case is so that the debug image topic is prefixed by the node name (so `/<node namespace>/<node name>/debug`). This is preferred for debugging information, but not necessarily for system-level messages like whether the race is over

Reply:
Using the private NodeHandle to get the node's name means that if we change the node name (in roslaunch or through refactoring), the name of debug image's topic gets updated automatically

Reply:
So the answer to "Should I change the subscriber to use nph too and remove the nh variable?" is no in my opinion

Reply:
Okay makes sense, thanks